setup.ps1

```
@@ -353,11 +355,108 @@ function Show-DirectAuthSecretNotice {
+    [void]$sb.Append("-----BEGIN $Type-----`n")
+    for ($i = 0; $i -lt $b64.Length; $i += 64) {
+        $len = [Math]::Min(64, $b64.Length - $i)
+        [void]$sb.Append($b64.Substring($i, $len))
+        [void]$sb.Append("`n")
+    }
+    [void]$sb.Append("-----END $Type-----`n")
+    [System.IO.File]::WriteAllText($Path, $sb.ToString())
+}
+
+# Generate a self-signed cert/key PEM pair if absent, using .NET (no openssl needed on Windows).
+function New-SelfSignedCertPair {
```

PSScriptAnalyzer (1.25.0)

[warning] Function 'New-SelfSignedCertPair' has verb that could change system state. Therefore, the function has to support 'ShouldProcess'.

(PSUseShouldProcessForStateChangingFunctions)